test/foundry/src/concrete/vault/OffchainAssetReceiptVault.withdraw.t.sol

```diff
@@ -0,0 +1,483 @@
+
+        // Stop the prank
+        vm.stopPrank();
+    }
+
+    /// Test withdraw reverts on InvalidId when id is 0
+    function testWithdrawInvalidId(
+        uint256 fuzzedKeyAlice,
+        uint256 fuzzedKeyBob,
+        uint256 assets,
+        uint256 minShareRatio,
+        bytes memory data,
+        string memory assetName,
+        string memory assetSymbol
+    ) external {
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        address alice = vm.addr((fuzzedKeyAlice % (SECP256K1_ORDER - 1)) + 1);
+        address bob = vm.addr((fuzzedKeyBob % (SECP256K1_ORDER - 1)) + 1);
+        minShareRatio = bound(minShareRatio, 0, 1e18);
+        // Assume that assets is not 0
+        assets = bound(assets, 1, type(uint256).max);
+
+        OffchainAssetReceiptVault vault = createVault(alice, assetName, assetSymbol);
+
+        // Prank as Alice to grant roles
+        vm.startPrank(alice);
+
+        vault.grantRole(vault.DEPOSITOR(), bob);
+        vault.grantRole(vault.WITHDRAWER(), bob);
+
+        // Prank Bob for the transaction
+        vm.startPrank(bob);
+
+        // Call the deposit function
+        vault.deposit(assets, bob, minShareRatio, data);
+
+        checkNoBalanceChange(vault, bob, bob, 0, assets, data, abi.encodeWithSelector(InvalidId.selector, 0));
+
+        // Stop the prank
+        vm.stopPrank();
+    }
+
+    /// Test withdraw function reverts when withdrawing someone else's assets
+    function testWithdrawOfSomeoneElse(
+        uint256 fuzzedKeyAlice,
+        uint256 fuzzedKeyBob,
+        uint256 assets,
+        uint256 minShareRatio,
+        bytes memory data,
+        string memory assetName,
+        string memory assetSymbol,
+        uint256 certifyUntil,
+        uint256 referenceBlockNumber,
+        bool forceUntil
+    ) external {
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        address alice = vm.addr((fuzzedKeyAlice % (SECP256K1_ORDER - 1)) + 1);
+        address bob = vm.addr((fuzzedKeyBob % (SECP256K1_ORDER - 1)) + 1);
+
+        vm.assume(alice != bob);
+
+        referenceBlockNumber = bound(referenceBlockNumber, 1, block.number);
+        certifyUntil = bound(certifyUntil, 1, type(uint32).max);
+
+        minShareRatio = bound(minShareRatio, 0, 1e18);
+        // Assume that assets is not 0
+        assets = bound(assets, 1, type(uint256).max);
+
+        OffchainAssetReceiptVault vault = createVault(alice, assetName, assetSymbol);
+
+        // Prank as Alice to set roles
+        vm.startPrank(alice);
+
+        vault.grantRole(vault.DEPOSITOR(), bob);
+        vault.grantRole(vault.WITHDRAWER(), bob);
+        vault.grantRole(vault.CERTIFIER(), bob);
+
+        // Prank Bob for the transaction
+        vm.startPrank(bob);
+
+        // Certify
+        vault.certify(certifyUntil, referenceBlockNumber, forceUntil, data);
+
+        // Call the deposit function
+        vault.deposit(assets, alice, minShareRatio, data);
```

Comment:
we should also check that when alice deposits her own assets, bob cannot withdraw them